tests/testthat/test-sanitize-filename.R

```diff
@@ -0,0 +1,23 @@
+context("sanitize-filename")
+
+test_that("File names can't have multiple dots", {
+  expect_equal(sanitizeFileName("../baz.jpg", "default"), ".baz.jpg")
+})
+
+test_that("If the file name is longer than 255 characters, the default name is returned", {
+  longName <- paste(replicate(256, "x"), collapse = "")
+  expect_equal(sanitizeFileName(longName, "default"), "default")
+})
+
+test_that("If the file name is empty, the default name is returned", {
+  expect_equal(sanitizeFileName("", "default"), "default")
+})
+
+test_that("Sanitized file names contain only alphanumeric characters and dots", {
+  crazyString <- " ℍ ℎ f ℏ ▅ ▆ ▇ ℐ oℑ ℒ  裸 邏 ℓo ⌔ ⌕ ⌖b ⌗ ⌘ar.csv"
```

Comment:
Using non-ASCII can cause problems on Windows, because the default encoding is not UTF-8. (Handling character encodings in R is kind of a pain.) For the htmltools package we did have some tests with extended characters, and this is what we needed to do to make them work: https://github.com/rstudio/htmltools/blob/02678ee/tests/testthat/test-template.R#L37-L38